texastribune/donations
# Reorg project structure

#### What's this PR do?
Reorganizes the donations app to...
* follow the "simple package" project structure mentioned [here](https://flask.palletsprojects.com/en/2.3.x/patterns/packages/#simple-packages)
* utilize docker-compose to clean up the Makefile

Also ...
* upgrades flask to 2.2.3
* pins requests to 2.29.0 because of a forced upgrade to urllib3 2.0 in 2.30.0 that was breaking our things

#### Why are we doing this? How does it help us?
* makes spinning the app up a simpler process
* makes writing tests easier and more modular
* makes the app pieces more modular as well
* adheres closer to our project structure in the texastribune repo, adding consistency
* hopefully this all adds up to a simpler flow for engineers when needing to update or add functionality to the app

#### How should this be manually tested?
You can test locally or on [test](https://donations-testing.texastribune.org/donate)

If testing locally, first:
* ensure your local env-docker file has the const `SLACK_CHANNEL` set to `#tech-test`
* run `make`

Either way:
* test making a donation (should show up in slack -> #tech-test if testing locally and #stripe if on test)

#### How should this change be communicated to end users?
Shouldn't need to be as the updates here are structural.

#### Are there any smells or added technical debt to note?
Now that our project structure is more advanced, there is more work we can do here like...
* move vue pieces out of /server/static/js and into their own folder at root ('client' or 'frontend')
* eventually modularize the app.py file by moving the routes to a views.py file and consider moving some of the other functions to utils or elsewhere, as appropriate

#### What are the relevant tickets?
https://airtable.com/appyo1zuQd8f4hBVx/tbloNZu8GkM52NKFR/viwS1XPty68eK4Ett/recm2EZHmnXTRlB2b?blocks=hide

#### Have you done the following, if applicable:
***(optional: add explanation between parentheses)***


#### TODOs / next steps:

* deploy to production
